### ISSUE TITLE
Response headers must appear in the documentation of the sync protocol

### ISSUE THREAD

**Issue by labkode** [cernbox/smashbox#40]
@moscicki @dragotin 

In the spec of the sync protocol we have to show the response headers in the examples, as they are a core part of the protocol.

An example: If you don't reply with the ETag on a download then the client will complain: https://github.com/owncloud/client/blob/7fc7925d17df58e6dfb9eb1383f0895d5fadce58/src/libsync/propagatedownload.cpp#L150

**Comment by dragotin** [comment]
Well, the ETag response header is mentioned in the file download section. The example there is very brief indeed, do you think that should be changed? 

Apart from that, I think the ETag is explained properly, or not?

**Comment by labkode** [comment]
@dragotin Yes, Klass, sorry for the noise. It is well explained in the file download section.

I have other question about the spec, the documentation just tell you the format for the calls/operations that the sync client does, but there is no information about the order of the operations.

I think it is important to document the sequence of events. 

For example, to setting up a working connection:
1. Issue a status.php
2. Ask for capabilities. TBD
3. Ask for quota. TBD
4. Ask for modification of top level. TBD
5. Connection validation call . TBD

I am not sure about the sequence of the TBD yet, but I am sure that if status.php fails, then you cannot continue.

What do you think ?

**Comment by dragotin** [comment]
@labkode absolutely, you're right. That needs to be defined as well, if there are constraints about the sequence. I will note that down as a TODO.

**Comment by labkode** [comment]
@dragotin Thanks, if I get some time I'll do it myself
